lib/fs.js

```diff
@@ -101,6 +101,10 @@ function nullCheck(path, callback) {
   return true;
 }
 
+function isFd(path) {
+  return (path >>> 0) === path && path >= 0;
```

Comment:
I'm pondering how much coercion is allowed. e.g. should we throw on `NaN`? my instinct is yes. all non uint32 values should throw. Which is exactly what you have now. :)